rfc-176-replacing-static.md

```diff
@@ -0,0 +1,127 @@
+
+### Recruitment Banners (move to `govuk_web_banners`)
+
+This isn’t a problem with `static`, but fits the banner theme and provides a neat start to the banner gem. Recruitment banners present a problem in that they often need to be inserted into the body of a page inside the area that `static` normally considers the app’s section (the wrapper). This means they need to be maintained in multiple apps.
+
+Our proposal is that we include the current duplicated recruitment banner config, code, and partial views in the gem. Then we can include the gem in each application and put a single call to it in the relevant app layouts. This means we can keep the configuration in one place, where it’s more easily checked for problems, and a single release of the banner gem will be automatically pulled into each relevant app at the next dependabot merge.
+
+### Static Assets (move to `govuk_web_layout`)
+
+#### How GOV.UK handles assets
+
+GOV.UK uses the standard rails assets pipeline for creation of assets, in which at asset compile time a fingerprint is generated for each asset based on the asset’s content, and during HTML page rendering the base names of the assets are replaced with the fingerprinted version. This allows clients to implement a pretty simple caching policy - they can always cache a given asset for as long as they like, because the asset’s name will change if the content changes - if the client doesn’t already have that fingerprinted file, they will need to download it.
+
+Asset pre-compilation happens when the container image is built (during the ‘deploy’ GitHub Actions workflow). This process generates the compiled assets and the Sprockets manifest. The manifest is the mechanism for Rails to map the original asset name (`crown.png`) to the fingerprinted asset name (`crown-b2ba232ade53223.png`), so the deployed app has a record of everything that was compiled as part of its assets:precompile step.
+
+Upon deployment into the Kubernetes cluster, a job runs which copies anything from `/app/public/assets/<name of app>` into an S3 bucket `s3://govuk-app-assets-<environment>/assets/<name of app>`. [Fastly maps] any call for (e.g.) `https://www.gov.uk/assets/` to that bucket, so a request for `https://www.gov.uk/assets/frontend/crown-b2ba232ade53223.png` will be served from that s3 bucket.
+
+
+![Current asset deploy pathway](rfc-176/current_govuk_asset_pathway.png)
+
+#### What this means for shared assets
+Because the upload-assets job in [govuk-helm-charts] includes the app name, there currently isn’t a mechanism for apps to upload assets in a shared context. If two apps both precompile a CSS file for the Table component, for instance, the fingerprint will be the same, but the path will be different, so a client that views two pages with Tables on (one in [frontend], one in [government-frontend]) will download the same file twice.
```

Comment:
Original intention was that it would act to _configure_ the app to share assets, but I think now there's just a much simpler way to do that (see my response to your comment further down).

Reply:
Marking as resolved as shared assets is now out of scope for this RFC.